### PR TITLE
Add keyboard 'UM-80'

### PR DESCRIPTION
**Pull request by duoshock**
## Description

Please add [UM-80](https://mergedesign.store/products/um-80) split keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/merge/um80/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+ *   .---.  |-------------------------.          '---------------------------------|        |---|
+ *   |M0 |  |CapsL |  A|  S|  D|  F|  G|          |  H|  J|  K|  L|  ;|  :|   Retn |        |End|
+ *   |---|  |----------------------------.        '--------------------------------|  .---. '---'
+ *   |M1 |  |Shft    |  Z|  X|  C|  V|  B|          |  N|  M|  ,|  ,|  /|  Shift|     |Up |
+ *   |---|  |----------------------------|          |---------------------------' .-----------.
+ *   |M2 |  |Ctl |Gui |Alt |Fn0 |Space   |          |    Space|   Alt|   Ctl|     |Lef|Dow|Rig|
+ *   '---'  '----------------------------'          '-----------------------'     '-----------'
+ */
+
+[_BASE] = LAYOUT_rspace_2u_bksp(
```
Comment (zvecr) [inline keyboards/merge/um80/keymaps/default/keymap.c:43]: This wont compile as the LAYOUT macro doesnt exit in `um80.h`

File: keyboards/merge/um80/keymaps/via/config.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * 
+ * This program is distributed in the hope that it will be useful, 
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+ * GNU General Public License for more details. 
+ * 
+ * You should have received a copy of the GNU General Public License 
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (zvecr) [inline keyboards/merge/um80/keymaps/via/config.h:16]:
```suggestion

#pragma once

```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/merge/um80/keymaps/default/keymap.c (hunk 1 above)
Comment (duoshock) [reply keyboards/merge/um80/keymaps/default/keymap.c]: I've just updated it. Thanks!

File: keyboards/merge/um80/config.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#ifdef RGB_MATRIX_ENABLE
+#    define DRIVER_LED_TOTAL RGBLED_NUM
+#    define RGB_MATRIX_SPLIT RGBLED_SPLIT
+#endif
+#define RGBLIGHT_SLEEP
+#define RGBLIGHT_LIMIT_VAL 120
+
+#define DEBOUNCE 5
+
+#define B6_AUDIO
```
Comment (drashna) [inline keyboards/merge/um80/config.h:53]:
```suggestion
#define AUDIO_PIN B6
```

File: keyboards/merge/um80/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    } else if (index == 1) { /* Master Right */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/merge/um80/keymaps/default/keymap.c:88]:
```suggestion
```

File: keyboards/merge/um80/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    } else if (index == 1) { /* Master Right */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/merge/um80/keymaps/via/keymap.c:88]:
```suggestion
```

File: keyboards/merge/um80/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# UM-80
+
+![UM-80](https://i.imgur.com/JVEC4Ntl.jpeg)
+
+UM-80 is a split keyboard in 75% layout with RGB, macro keys, OLED screen and encoder housed in PLA 3D printed cases sold by Merge. [Product page](https://mergedesign.store/products/um-80)
+
+* Keyboard Maintainer: [duoshock](https://github.com/duoshock)
+* Hardware Availability: [Merge Store](https://mergedesign.store/products/um-80)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/merge/um80/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/merge/um80/readme.md (hunk 6 above)
Comment (duoshock) [reply keyboards/merge/um80/readme.md:10]: Sure. I've just added it. Thanks!

**Review by drashna — APPROVED** [review]

**Comment by duoshock** [comment]
Sorry to chase. Please let me know if there is any amendment needed; I will get that fixed. :)

File: keyboards/merge/um80/um80.c [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    // WPM counter Start
+    oled_set_cursor(0,13);
+    oled_write(" ", false);
+    oled_write(get_u8_str(get_current_wpm(), '0'), false);
+    oled_set_cursor(0,14);
+    oled_write(" WPM ", false);
+    // WPM counter End
+}
+
+bool oled_task_user(void) {
```
Comment (zvecr) [inline keyboards/merge/um80/um80.c:108]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) { return false; }
```

File: keyboards/merge/um80/um80.c [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+ */
+
+#include "um80.h"
+
+#ifdef OLED_ENABLE
+void suspend_power_down_user(void) {
+    oled_off();
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```
Comment (zvecr) [inline keyboards/merge/um80/um80.c:24]:
```suggestion
oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
```

File: keyboards/merge/um80/um80.c [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+ * You should have received a copy of the GNU General Public License 
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "um80.h"
+
+#ifdef OLED_ENABLE
+void suspend_power_down_user(void) {
+    oled_off();
+}
```
Comment (zvecr) [inline keyboards/merge/um80/um80.c:22]:
```suggestion
void suspend_power_down_kb(void) {
    oled_off();
    suspend_power_down_user();
}
```

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**